src/locate/synctex/worker.ts

```diff
@@ -91,32 +92,37 @@ function toRect(blocks: any): Rectangle {
     }
 }
 
-function parseSyncTexForPdf(pdfFile: string): PdfSyncObject | undefined {
-    const filename = path.basename(pdfFile, path.extname(pdfFile))
-    const dir = path.dirname(pdfFile)
-    const synctexFile = path.resolve(dir, filename + '.synctex')
-    const synctexFileGz = synctexFile + '.gz'
+async function parseSyncTexForPdf(pdfFile: vscode.Uri): Promise<PdfSyncObject | undefined> {
+    const filename = path.basename(pdfFile.fsPath, path.extname(pdfFile.fsPath))
+    const dir = path.dirname(pdfFile.fsPath)
+    const synctexFile = vscode.Uri.parse(pdfFile.scheme + "://" + path.resolve(dir, filename + '.synctex'))
+    const synctexFileGz = vscode.Uri.parse(pdfFile.scheme + "://" + synctexFile.fsPath + '.gz')
 
-    if (fs.existsSync(synctexFile)) {
+    try {
+        await vscode.workspace.fs.stat(synctexFile);
         try {
             logger.log(`Parsing .synctex ${synctexFile} .`)
-            const s = fs.readFileSync(synctexFile, {encoding: 'binary'})
-            return parseSyncTex(s)
+            const s = await vscode.workspace.fs.readFile(synctexFile)
+            return parseSyncTex(new TextDecoder().decode(s))
```

Comment:
Why decode this?

Reply:
Just applied to parseSyncTex(pdfsyncBody: string)

Reply:
Use `lw.file.read`